src/bootstrap/src/bin/main.rs

```diff
@@ -132,6 +140,8 @@ fn check_version(config: &Config) -> Option<String> {
             msg.push_str(&format!(
                 "update `config.toml` to use `change-id = {latest_config_id}` instead"
             ));
+
+            t!(fs::write(warned_id_path, id.to_string()));
```

Comment:
Can you detect if this is running in a terminal or not? When it is not in a terminal we probably don't want to update this file.

Reply:
> "running in a terminal or not"
You mean if it's CI environment or not?

Reply:
No, I mean running in a terminal (as in the `is_terminal` function). x.py is also often executed by rust-analyzer, to give one example.

Reply:
Got it, thanks.

https://github.com/rust-lang/rust/pull/117815/commits/89829bb83215799e8c99bdda0b753b59e73aa119 should do the work.